R/ResampleInstances.R

```diff
@@ -1,54 +1,64 @@
-instantiateResampleInstance = function(desc, size, task, coords) {
+instantiateResampleInstance = function(desc, size, task) {
   UseMethod("instantiateResampleInstance")
 }
 
-instantiateResampleInstance.HoldoutDesc = function(desc, size, task = NULL, coords) {
+instantiateResampleInstance.HoldoutDesc = function(desc, size, task = NULL) {
   inds = sample(size, size * desc$split)
   makeResampleInstanceInternal(desc, size, train.inds = list(inds))
 }
 
-instantiateResampleInstance.CVDesc = function(desc, size, task = NULL, coords) {
+instantiateResampleInstance.CVDesc = function(desc, size, task = NULL) {
   if (desc$iters > size)
     stopf("Cannot use more folds (%i) than size (%i)!", desc$iters, size)
   test.inds = chunk(seq_len(size), shuffle = TRUE, n.chunks = desc$iters)
   makeResampleInstanceInternal(desc, size, test.inds = test.inds)
 }
 
-instantiateResampleInstance.SpCVDesc = function(desc, size, task = NULL, coords = NULL) {
-  if (is.null(coords)) {
-    coords = data.frame(task$env$data$x, task$env$data$y)
-  } else {
-    coords = coords
+instantiateResampleInstance.SpCVDesc = function(desc, size, task = NULL) {
+
+  if (is.null(task)) {
+    stopf("Please provide a task.")
+  }
+  if (is.null(task$coordinates)) {
+    stopf("Please provide suitable coordinates for SpCV. See ?Task for help.")
   }
-  # perform kmeans clustering
-  inds = kmeans(coords, centers = desc$iters)
-  inds = factor(inds$cluster)
+  else {
```

Comment:
Unnecessary nesting, since `stopf` will end the program flow in any case.
Besides, we do
```R
} else {
```
instead of
```R
}
else {
```

Reply:
Ok, removed it.